R/opendp/src/core.c

```diff
@@ -0,0 +1,502 @@
+// Auto-generated. Do not edit.
+#include <R.h>
+#include <Rmath.h>
+#include <R_ext/Boolean.h>
+#include <R_ext/Rdynload.h>
+#include <Rdefines.h>
+#include <Rinternals.h>
+#include <R_ext/Complex.h>
+
+#include "convert.h"
+#include "convert_elements.h"
+#include "Ropendp.h"
+#include "opendp.h"
+#include "opendp_extras.h"
+
+
+SEXP core__function_eval(
+    SEXP this, SEXP arg, SEXP TI, SEXP T_arg, SEXP log
```

Comment:
Why both TI and T_arg?

Reply:
The bootstrap macro is annotating an extern "C" function that doesn't have generics. An extra unused TI is added on the rust side to help get the types sorted that is promptly discarded. Not a problem with R codegen.